src/components/DialogConfigButton.js

```diff
@@ -0,0 +1,124 @@
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Typography,
+  TextField,
+  Box
+} from '@mui/material'
+import SettingsIcon from '@mui/icons-material/Settings'
+import React, { useState } from 'react'
+import {
+  createChromeAlarm,
+  INITIAL_CHECK_INTERVAL_TIMER_MINUTES
+} from '../background'
+import { useChromeLocalStorage } from '../hooks/useChromeLocalStorage'
+
+const SETTINGS_LABELS = {
+  checkIntervalTimer: 'Check Interval Timer (minutes)'
+}
+
+const rowStyles = {
+  display: 'flex',
+  flexDirection: 'row',
+  alignItems: 'center',
+  justifyContent: 'space-around'
+}
+
+const ConfigurationTitle = ({ title }) => {
+  return (
+    <Box sx={{ maxWidth: '50%' }}>
+      <Typography fontSize={14}>{title}</Typography>
+    </Box>
+  )
+}
+
+const ConfigurationField = ({ renderField }) => {
+  return <Box sx={{ maxWidth: '20%' }}>{renderField()}</Box>
+}
+
+export default function DialogConfigButton({ title, disabled, ...props }) {
+  const [open, setOpen] = useState(false)
+  const {
+    setData: setCheckIntervalTimer,
+    localData: localCheckIntervalTimer,
+    setLocalData: setLocalCheckIntervalTimer,
+    hasError,
+    setHasError
+  } = useChromeLocalStorage(
+    'checkIntervalTimer',
+    INITIAL_CHECK_INTERVAL_TIMER_MINUTES
+  )
+
+  const handleClickOpen = () => setOpen(true)
+  const handleDialogClose = () => setOpen(false)
+
+  const handleCheckIntervalTimerValidation = value => {
+    setLocalCheckIntervalTimer(value.trim())
+    if (!Number.isInteger(Number(value))) {
```

Comment:
you can refactor this like:
```suggestion
   const hasError = !Number.isInteger(Number(value))
   setHasError(hasError)
```